polkadot/node/network/collator-protocol/src/validator_side/collation.rs

```diff
@@ -247,6 +322,8 @@ impl Collations {
 		&mut self,
 		finished_one: &(CollatorId, Option<CandidateHash>),
 		relay_parent_mode: ProspectiveParachainsMode,
+		group_assignments: &Vec<ParaId>,
+		pending_fetches: &BTreeMap<ParaId, usize>,
```

Comment:
This is actually pointless since we are not fetching collations for the same relay parent in parallel. If there is a pending fetch or not can be deduced from `CollationStatus` and this code can be simplified.